src/main/java/marquez/service/JobService.java

```diff
@@ -196,8 +200,12 @@ public Job createOrUpdate(
         }
         final JobRow jobRow = jobDao.find(namespaceName.getValue(), jobName.getValue()).get();
         final JobContextRow contextRow = jobContextDao.findBy(checksum).get();
-        final List<UUID> inputUuids = findUuids(jobMeta.getInputs());
-        final List<UUID> outputUuids = findUuids(jobMeta.getOutputs());
+        final List<DatasetRow> inputRows = findUuids(jobMeta.getInputs());
+        final List<UUID> inputUuids =
+            inputRows.stream().map(e -> e.getUuid()).collect(Collectors.toList());
+        final List<DatasetRow> outputRows = findUuids(jobMeta.getOutputs());
```

Comment:
_Minor:_ Given the method now returns _row_ objects, I'd rename to `findDatasetRows()`